File: LeetCode/LeetCode Practice Problems/1614- Maximum Nesting Depth of the Parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int cnt = 0 , mx = 0;
        for(char c: s) {
            if(c == '(') {
                cnt++;
            } else if(c == ')') {
                cnt--;
            }
            
            mx = max(mx , cnt);
        }
        
        return mx;
    }
};

File: LeetCode/LeetCode Practice Problems/1614-Maximum Nesting Depth of the Parentheses.java
class Solution {
    public int maxDepth(String s) {
        int ans = 0;
        int cur = 0;
        
        for(char c: s.toCharArray()) {
            if(c == '(') {
                cur++;
                ans = Math.max(ans , cur);
            } else if(c == ')') {
                cur--;
            }
        }
        
        return ans;
    }
}
